docker.container panics if image not present

I ran`plan` for the dockerContainer.hcl sample after force removing the nginx:1.10-alpine image, and got this output. After successfully pulling the image, I ran `plan` for dockerContainer.hcl and again got the error.
```
./converge plan --local samples/dockerContainer.hcl
2016-12-07T09:13:37-06:00 |WARN| setting session-local token	token=6ee22f4d-2d11-4dff-bb3a-6d61c3c2db04
2016-12-07T09:13:37-06:00 |WARN| no SSL config in use, server will accept unencrypted connections	component=rpc
2016-12-07T09:13:37-06:00 |INFO| serving GRPC	addr=http://127.0.0.1:47740 component=rpc
2016-12-07T09:13:37-06:00 |INFO| serving REST	addr=http://127.0.0.1:47740 component=rpc
2016-12-07T09:13:37-06:00 |WARN| skipping module verification	component=client
2016-12-07T09:13:37-06:00 |INFO| got status	component=client file=samples/dockerContainer.hcl id=root/docker.container.nginx run=STARTED stage=PLAN
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x250 pc=0x737e1c]

goroutine 83 [running]:
panic(0xa5d800, 0xc4200100e0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/asteris-llc/converge/resource/docker/container.(*Container).diffContainer(0xc420124180, 0xc420186e00, 0xc42026aa80, 0xc4202ad8d8, 0xc4202c5200)
	/vagrant/godev/src/github.com/asteris-llc/converge/resource/docker/container/container.go:195 +0x9fc
github.com/asteris-llc/converge/resource/docker/container.(*Container).Check(0xc420124180, 0x1065020, 0xc420246db0, 0x10629e0, 0xc420282980, 0x1b, 0xc420282980, 0x0, 0x0)
	/vagrant/godev/src/github.com/asteris-llc/converge/resource/docker/container/container.go:84 +0x39b
github.com/asteris-llc/converge/resource.(*TaskWrapper).Check(0xc420278790, 0x1065020, 0xc420246db0, 0x10629e0, 0xc420282980, 0x0, 0x7fb8899fd9c0, 0xc420278790, 0xa57c20)
	<autogenerated>:25 +0x75
github.com/asteris-llc/converge/plan.(*pipelineGen).PlanNode(0xc420285060, 0x1065020, 0xc420246db0, 0xa733c0, 0xc420279740, 0xa733c0, 0xc420279740, 0x0, 0x0)
	/vagrant/godev/src/github.com/asteris-llc/converge/plan/pipeline.go:163 +0x1f0
github.com/asteris-llc/converge/plan.(*pipelineGen).PlanNode-fm(0x1065020, 0xc420246db0, 0xa733c0, 0xc420279740, 0xa733c0, 0xc420279740, 0x0, 0x0)
	/vagrant/godev/src/github.com/asteris-llc/converge/plan/pipeline.go:49 +0x52
github.com/asteris-llc/converge/executor.Pipeline.Exec(0xc420285080, 0x4, 0x4, 0x1065020, 0xc420246db0, 0xa6dc60, 0xc420278790, 0x4, 0x4, 0x58db11, ...)
	/vagrant/godev/src/github.com/asteris-llc/converge/executor/pipeline.go:53 +0x7b
github.com/asteris-llc/converge/plan.WithNotify.func1(0xc420282000, 0xc420286ed0, 0x0, 0xc420284801)
	/vagrant/godev/src/github.com/asteris-llc/converge/plan/plan.go:49 +0x127
github.com/asteris-llc/converge/graph.NotifyPre.func1(0xc420282000, 0xc420286ed0, 0xc420221cb0, 0x48b6a9)
	/vagrant/godev/src/github.com/asteris-llc/converge/graph/notifier.go:29 +0x6b
github.com/asteris-llc/converge/graph.NotifyPost.func1(0xc420282000, 0xc420286ed0, 0xc420282000, 0xc420221e01)
	/vagrant/godev/src/github.com/asteris-llc/converge/graph/notifier.go:36 +0x49
github.com/asteris-llc/converge/graph.transform.func1(0xc420282000, 0xc420260320, 0x1b)
	/vagrant/godev/src/github.com/asteris-llc/converge/graph/graph.go:697 +0x3b
github.com/asteris-llc/converge/graph.dependencyWalk.func5(0xc420260320, 0x1b)
	/vagrant/godev/src/github.com/asteris-llc/converge/graph/graph.go:464 +0xb08
created by github.com/asteris-llc/converge/graph.dependencyWalk.func3
	/vagrant/godev/src/github.com/asteris-llc/converge/graph/graph.go:378 +0x4f6
```

--- a/resource/docker/container/container.go
+++ b/resource/docker/container/container.go
@@ -112,7 +112,10 @@ func (c *Container) Check(context.Context, resource.Renderer) (resource.TaskStat
 	if container != nil {
 		status.AddDifference("name", strings.TrimPrefix(container.Name, "/"), c.Name, "")
 		if c.Force {
-			c.diffContainer(container, status)
+			if diffErr := c.diffContainer(container, status); diffErr != nil {
+				return nil, diffErr
+			}
+			return nil, err
 		}
 	} else {
 		status.AddDifference("name", "", c.Name, "<container-missing>")
@@ -217,6 +220,9 @@ func (c *Container) diffContainer(container *dc.Container, status *resource.Stat
 	if err != nil {
 		status.Level = resource.StatusFatal
 		return errors.Wrapf(err, "failed to find image %s for container %s", container.Image, container.Name)
+	}
+	if image == nil {
+		return errors.New("backing image is unavailable")
 	}
 
 	var actual, expected string

--- a/resource/docker/container/container.go
+++ b/resource/docker/container/container.go
@@ -115,7 +115,6 @@ func (c *Container) Check(context.Context, resource.Renderer) (resource.TaskStat
 			if diffErr := c.diffContainer(container, status); diffErr != nil {
 				return nil, diffErr
 			}
-			return nil, err
 		}
 	} else {
 		status.AddDifference("name", "", c.Name, "<container-missing>")

--- a/resource/docker/container/container_test.go
+++ b/resource/docker/container/container_test.go
@@ -28,6 +28,7 @@ import (
 	"github.com/asteris-llc/converge/resource/docker/container"
 	dc "github.com/fsouza/go-dockerclient"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"golang.org/x/net/context"
 )
 
@@ -94,7 +95,7 @@ func TestContainerCheck(t *testing.T) {
 
 		status, err := container.Check(context.Background(), fakerenderer.New())
 
-		assert.Nil(t, err)
+		require.NoError(t, err)
 		assert.False(t, status.HasChanges())
 	})
 

--- a/resource/docker/container/container_test.go
+++ b/resource/docker/container/container_test.go
@@ -99,6 +99,27 @@ func TestContainerCheck(t *testing.T) {
 		assert.False(t, status.HasChanges())
 	})
 
+	t.Run("missing image for container", func(t *testing.T) {
+		c := &fakeAPIClient{
+			FindContainerFunc: func(string) (*dc.Container, error) {
+				return &dc.Container{
+					Name:   "nginx",
+					State:  dc.State{Status: "running"},
+					Config: &dc.Config{}}, nil
+			},
+			FindImageFunc: func(string) (*dc.Image, error) {
+				return nil, nil
+			},
+		}
+
+		container := &container.Container{Force: true, Name: "nginx"}
+		container.SetClient(c)
+
+		_, err := container.Check(context.Background(), fakerenderer.New())
+
+		require.Error(t, err)
+	})
+
 	t.Run("status change", func(t *testing.T) {
 		c := &fakeAPIClient{
 			// the existing container is running the "nginx" command

